C/common/reading.cpp

```diff
@@ -225,6 +225,46 @@ ostringstream convert;
 	return convert.str();
 }
 
+/**
+ * Convert time since epoch to a formatted m_timestamp DataTime in UTC 
+ * and use a cache to speed it up
+ * @param tv_sec	Seconds since epoch
+ * @param date_time	Buffer in which to return the formatted timestamp
+ * @param dateFormat	Format: FMT_DEFAULT or FMT_STANDARD
+ */
+void Reading::getFormattedDateTimeStr(const time_t *tv_sec, char *date_time, readingTimeFormat dateFormat) const
+{
+	static unsigned long cached_sec_since_epoch = 0;
```

Comment:
The cache is not thread safe. This is a core library function that is used in all services, some of which have multiple threads that can be adding readings.